Cython/Compiler/PyrexTypes.py

```diff
@@ -4716,6 +4716,13 @@ class PythonTypeConstructorMixin:
     """Used to help Cython interpret indexed types from the typing module (or similar)
     """
     modifier_name = None
+    contains_none = False
+
+    def allows_none(self):
+        return (
+            self.modifier_name == 'typing.Optional' or (
+            self.modifier_name == 'typing.Union' and self.contains_none)
+        )
```

Comment:
The position of the parentheses here feel somewhat misleading to me. They are also not required. Let's just remove them and let the line separation work for us.
```suggestion
        return (
            self.modifier_name == 'typing.Optional' or
            self.modifier_name == 'typing.Union' and self.contains_none
        )
```